sdk/eventgrid/azure-eventgrid/tests/test_eg_publisher_client.py

```diff
@@ -343,5 +344,19 @@ def test_send_custom_schema_event_as_list(self, resource_group, eventgrid_topic,
 
     def test_send_throws_with_bad_credential(self):
         bad_credential = "I am a bad credential"
-        with pytest.raises(ValueError, match="The provided credential should be an instance of AzureSasCredential or AzureKeyCredential"):
+        with pytest.raises(ValueError, match="The provided credential should be an instance of a TokenCredential, AzureSasCredential or AzureKeyCredential"):
             client = EventGridPublisherClient("eventgrid_endpoint", bad_credential)
+
+    @pytest.mark.live_test_only
```

Comment:
why live only?

Reply:
little tricky to generate recording given we use resource group preparers and envvars for secrets